sdk/generic-sdk-core/generic-core/src/main/java/com/generic/core/http/models/RetryOptions.java

```diff
@@ -0,0 +1,141 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package com.generic.core.http.models;
+
+import com.generic.core.http.policy.RequestRetryCondition;
+import com.generic.core.models.HeaderName;
+import com.generic.core.util.ClientLogger;
+
+import java.time.Duration;
+import java.util.HashMap;
+import java.util.Objects;
+import java.util.function.Predicate;
+
+/**
+ * Options to configure the retry policy's behavior.
+ */
+public class RetryOptions {
+    private static final ClientLogger LOGGER = new ClientLogger(RetryOptions.class);
+    private int maxRetries;
```

Comment:
I think this could be made `final`, don't see it being mutated anywhere within the class